### PR TITLE
Cp airplane flight

### PR DESCRIPTION
**Pull request by cpatelss90**
Finished Airplane & Flight(DAO,Service,Controller) and added some tests

### CODE REVIEW

**Comment by Joshua-Podhola** [comment]
Merge conflicts fixed

**Comment by Joshua-Podhola** [comment]
Any particular reason the contents of bin/ are included in the pull?

File: src/main/java/com/smoothstack/utopiaairlines/UtopiaAirlinesApplication.java [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+//@ComponentScan(basePackages = {"com.smoothstack.utopiaairlines.entities"})
+
+
```
Comment (Joshua-Podhola) [inline src/main/java/com/smoothstack/utopiaairlines/UtopiaAirlinesApplication.java]: Why is this here?

File: src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	//Update Airplane ID by Flight ID
+	@PutMapping("/routes/flight/update/flight_id/airplane_id/{id}/{airplaneId}")
+	public String updateAirplaneIdByFlightId(@PathVariable Integer id, @PathVariable Integer airplaneId) {
+		flightService.updateAirplaneIdByFlightId(id, airplaneId);
+		return "Airplane ID Successfully Upated!";
+	}
+	
+	
+	//Update Departure Date Time by Flight ID
+	@PutMapping("/routes/flight/update/flight_id/departure_time/{id}/{dateTime}")
```
Comment (Joshua-Podhola) [inline src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java]: I won't make a fuss about this, but for the capstone project, consider using query parameters

File: src/main/java/com/smoothstack/utopiaairlines/dao/FlightDao.java [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
+@Transactional
 public interface FlightDao extends JpaRepository<Flight, Integer> {
+	
+	List<Flight> findByRouteId(Integer id);
+	List<Flight> findByAirplaneId(Integer id);
+		
+	@Query(
+			  value = "SELECT * FROM flight f WHERE f.departure_time = ?", 
+			  nativeQuery = true)
+	List<Flight> findByDT(LocalDateTime dateTime);
```
Comment (Joshua-Podhola) [inline src/main/java/com/smoothstack/utopiaairlines/dao/FlightDao.java]: Searching by exact departure time is a little silly; I think BETWEEN two dates or AFTER a certain date, or accept a Date instead:

SELECT *
FROM flight f
WHERE DATE(f.departure_time) = ?;

File: src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		Optional<Airplane> airplanes = airplaneDao.findById(id);
+		return airplanes;
+	}
+	
+	
+	// Update airplane By Id
+	public void updateAirplaneById(Integer id, Integer capacity){
+		Optional<Airplane> airplanes = findAirplaneById(id);
+		Airplane airplane = airplanes.get();
+		airplane.setId(id);
```
Comment (Joshua-Podhola) [inline src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java]: Why are you setting the ID here? If you're correctly retrieving the airplane from the database by its ID, the ID should be populated already, right? I feel like this might be indicative of potential bugs if this is needed.

File: src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import com.smoothstack.utopiaairlines.entities.Airplane;
+import com.smoothstack.utopiaairlines.entities.Flight;
+import com.smoothstack.utopiaairlines.service.AirplaneService;
+import com.smoothstack.utopiaairlines.service.FlightService;
+
+@RestController
+public class AdminController {
```
Comment (Joshua-Podhola) [inline src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java:25]: We should discuss potentially splitting the controllers up based on entity; having them all in one class seems like an organization nightmare.

File: src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	public Optional<Airplane> findAirplaneById(Integer id){
+		Optional<Airplane> airplanes = airplaneDao.findById(id);
+		return airplanes;
+	}
+	
+	
+	// Update airplane By Id
+	public void updateAirplaneById(Integer id, Integer capacity){
+		Optional<Airplane> airplanes = findAirplaneById(id);
+		Airplane airplane = airplanes.get();
```
Comment (Joshua-Podhola) [inline src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java]: Will throw NoSuchElementException if no airplane by that ID is found. This is never caught nor checked for, neither here nor in the controller.

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -1,4 +1,8 @@
...
-spring.datasource.username=utopia
-spring.datasource.password=NAg6K0V2guDxAXQC
-spring.datasource.url=jdbc:mysql://localhost:3306/utopia
+spring.datasource.username=root
+spring.datasource.password=Password
+spring.datasource.url=jdbc:mysql://localhost:3306/utopia_airlines
+
+
+spring.jpa.hibernate.ddl-auto = update
+
```
Comment (Joshua-Podhola) [inline src/main/resources/application.properties]: Please, please don't version the application.properties file if you can help it - we're going to end up conflicting here because everyone's database credentials are different.

**Review by Joshua-Podhola — COMMENTED** [review]
The only reason I'm not approving this quite yet is because a couple of things need to be discussed:
1) We need to make sure we're not fighting over the contents of the application.properties file
2) How much input validation and error checking are we doing?
3) We should probably un-version the bin folder; it's just wasted space.

I'm fine with fixing up some input validation, error checking, testing, and comments in a later pull, we just need to talk about that properties file.

File: src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java (hunk 2 above)
Comment (cpatelss90) [reply src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java]: Done

File: src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java (hunk 4 above)
Comment (cpatelss90) [reply src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java]: Done

File: src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java (hunk 5 above)
Comment (cpatelss90) [reply src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java:25]: Lets discuss this with everyone once we are ready to move on to next micro service

File: src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java (hunk 6 above)
Comment (cpatelss90) [reply src/main/java/com/smoothstack/utopiaairlines/service/AirplaneService.java]: Done.

**Comment by cpatelss90** [comment]
I am making a new pull req with fixed changes.

**Review by Joshua-Podhola — APPROVED** [review]
